File: KZP/lab3/src/ki304/rybka/lab3/MediaPlayer.java
/**
 * 
 */
package ki304.rybka.lab3;

/**
 * Class MediaPlayer implements media player
 * @author Severyn Rybka
 * @version 1.0
 * @since version 1.0
 */
public class MediaPlayer 
{
	private LogManager logManager;
	private boolean isPlaying = false;
	private int currentVideoIndex = 0;
	private int totalVideos = 0;
	private String[] videoNames;
	/**
	 * Constructor of MediaPlayer
	 * @param _logManager LogManager
	 */
	public MediaPlayer(LogManager _logManager)
	{
		logManager = _logManager;
	}
	/**
	 * Play
	 */
	public void play()
	{
		if (!isPlaying && totalVideos > 0)
		{
			isPlaying = true;
			logManager.info("Playing: " + videoNames[currentVideoIndex]);
		}
	}
	/**
	 * Pause
	 */
	public void pause()
	{
		if (isPlaying)
		{
			isPlaying = false;
			logManager.info("Paused");
		}
	}
	/**
	 * Next video
	 */
	public void nextVideo() 
	{
		if (currentVideoIndex < totalVideos-1)
		{
			currentVideoIndex++;
			logManager.info("Playing: " + videoNames[currentVideoIndex]);
			isPlaying = true;
			return;
		}
		logManager.error("No next video");
	}
	/**
	 * Previous video
	 */
	public void prevVideo() 
	{
		if (currentVideoIndex > 0)
		{
			currentVideoIndex--;
			logManager.info("Playing: " + videoNames[currentVideoIndex]);
			isPlaying = true;
			return;
		}
		
		logManager.error("No prev video");
	}
	/**
	 * Set video list
	 * @param videos Video names
	 */
	public void setVideos(String[] videos) 
	{
		videoNames = videos;
		totalVideos = videos.length;
		currentVideoIndex = 0;
		for (String name : videoNames)
		{
			logManager.info("Video added: " + name);
		}
		isPlaying = false;
	}
	/**
	 * Show current video
	 */
	public void showCurrentVideo()
	{
		if (totalVideos > 0)
			logManager.info("Current: " + videoNames[currentVideoIndex]);
	}
	/**
	 * Show videos
	 */
	public void showVideos()
	{
		for (String name : videoNames)
		{
			logManager.info("Video: " + name);
		}
	}
}

File: KZP/lab9/media_player.py
"""
MediaPlayer
"""
from log_manager import LogManager

class MediaPlayer:
    def __init__(self, log_manager):
        """ Constructor """
        self.__logManager = log_manager
        self.__isPlaying = False
        self.__currentVideoIndex = 0
        self.__totalVideos = 0
        self.__videoNames = []


    def play(self):
        """ Play video """
        if self.__isPlaying == False and self.__totalVideos > 0:
            self.__isPlaying = True
            self.__logManager.info(f"Playing: {self.__videoNames[self.__currentVideoIndex]}")


    def pause(self):
        """ Pause video """
        if self.__isPlaying == True:
            self.__isPlaying = False
            self.__logManager.info("Paused")


    def nextVideo(self):
        """ Next video """
        if self.__currentVideoIndex < self.__totalVideos - 1:
            self.__currentVideoIndex += 1
            self.__isPlaying = True
            self.__logManager.info(f"Playing: {self.__videoNames[self.__currentVideoIndex]}")
        else:
            self.__logManager.error("No next video")


    def prevVideo(self):
        """ Previous video """
        if self.__currentVideoIndex > 0:
            self.__currentVideoIndex -= 1
            self.__isPlaying = True
            self.__logManager.info(f"Playing: {self.__videoNames[self.__currentVideoIndex]}")
        else:
            self.__logManager.error("No prev video")


    def setVideos(self, videos):
        """ Set videos """
        self.__videoNames = videos
        self.__totalVideos = len(self.__videoNames)
        self.__currentVideoIndex = 0
        self.__isPlaying = False
        for video in self.__videoNames:
            self.__logManager.info(f"Video added: {video}")


    def showCurrentVideo(self):
        """ Show current video name """
        if self.__totalVideos > 0:
            self.__logManager.info(f"Current: {self.__videoNames[self.__currentVideoIndex]}")


    def showVideos(self):
        """ Show video names """
        for video in self.__videoNames:
            self.__logManager.info(f"Video: {video}")
            